### PR TITLE
Fix HU locale test for ICU

### PR DESCRIPTION
**Pull request by steveharter**
Fixed test for Hungarian contractions + ICU. See https://github.com/dotnet/corefx/issues/3972 for more detail

@stephentoub

### CODE REVIEW

File: src/System.Runtime/tests/System/String.cs [hunk 1]
```diff
@@ -788,6 +795,7 @@ public static void TestIndexOf_EquivalentDiacritics()
             Assert.Equal(8, source.IndexOf(target, StringComparison.CurrentCultureIgnoreCase));
         });
 
+        target = "\u0300";
```
Comment (eerhardt) [inline src/System.Runtime/tests/System/String.cs]: Is this intended?

File: src/System.Runtime/tests/System/String.cs [hunk 2]
```diff
@@ -751,10 +753,15 @@ public static void TestIndexOf_HungarianDoubleCompression()
...
-            //Assert.Equal(0, source.IndexOf(target, StringComparison.CurrentCulture));
+            /* 
+             There are differences between Windows and ICU regarding contractions.
+             Windows has equal contraction collation weights, including case (target="Ddzs" same behavior as "ddzs").
+             ICU has different contraction collation weights, depending on locale collation rules.
+             If CurrentCultureIgnoreCase is specified, ICU will use 'secondary' collation rules
+              which ignore the contraction collation weights (defined as 'tertiary' rules)
+            */
+            Assert.Equal(s_isWindows ? 0 : -1, source.IndexOf(target));
+            Assert.Equal(s_isWindows ? 0 : -1, source.IndexOf(target, StringComparison.CurrentCulture));
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/String.cs]: Is there a corresponding test for non-Windows we could add that would test the positive case?

File: src/System.Runtime/tests/System/String.cs (hunk 1 above)
Comment (steveharter) [inline src/System.Runtime/tests/System/String.cs]: No. Odd. Perhaps a merge issue

File: src/System.Runtime/tests/System/String.cs (hunk 2 above)
Comment (steveharter) [inline src/System.Runtime/tests/System/String.cs]: The next line (not shown) does test for the positive case:
Assert.Equal(0, source.IndexOf(target, StringComparison.CurrentCultureIgnoreCase));

Comment (stephentoub) [inline src/System.Runtime/tests/System/String.cs]: Ah, sorry, it was hidden in the diff.  Thanks.

**Pull request merged**